server/master/src/main/java/org/apache/accumulo/master/tableOps/TraceRepo.java

```diff
@@ -72,4 +75,19 @@ public String getReturn() {
     return repo.getReturn();
   }
 
+  /**
+   * @return string version of Repo that is suitable for logging
+   */
+  public static String toLogString(Repo<Master> repo) {
+    if (repo instanceof TraceRepo) {
+      // There are two reasons the repo is unwrapped. First I could not figure out how to get this
+      // to work with Gson. Gson kept serializing nothing for the generic pointer TraceRepo.repo.
+      // Second I thought this information was not useful for logging.
+      repo = ((TraceRepo<Master>) repo).repo;
```

Comment:
Memory may not be serving me well but I think you can define a JSONSerializer that may help avoid this [1]. But I could be wrong as it's much easier to be an armchair quarterback reviewer than the one writing the code :)

[1] https://howtodoinjava.com/gson/custom-serialization-deserialization/

Reply:
I think I tried some of those.  My solutions kept leading to an infinite loop or no output.  The wall I kept running into was that Gson wanted to work with concrete types. I had a generic interface that I wanted Gson to generate json for.  Gson did not like that, it wants to write out things that could be deserialized.  I was trying to get it write json out that it would not be able to deserialize back into concrete types.